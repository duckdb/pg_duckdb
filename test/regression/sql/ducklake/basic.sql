CREATE TABLE t(a INT) USING ducklake;
INSERT INTO t SELECT g % 10 FROM generate_series(1, 1000) g;

SELECT COUNT(*) FROM t;
SELECT a, COUNT(*) FROM t WHERE a > 5 GROUP BY a ORDER BY a;


CREATE TABLE h(a INT);
INSERT INTO h SELECT g % 10 FROM generate_series(1, 1000) g;

-- insert into ducklake from an heap table
INSERT INTO t SELECT * FROM h;
SELECT COUNT(*) FROM t;

DROP TABLE t;
DROP TABLE h;

-- empty table
CREATE TABLE empty(a INT) USING ducklake;
SELECT COUNT(*) FROM empty;
DROP TABLE empty;
