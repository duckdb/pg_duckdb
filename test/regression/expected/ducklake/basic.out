CREATE TABLE t(a INT) USING ducklake;
INSERT INTO t SELECT g % 10 FROM generate_series(1, 1000) g;
SELECT COUNT(*) FROM t;
 count 
-------
  1000
(1 row)

SELECT a, COUNT(*) FROM t WHERE a > 5 GROUP BY a ORDER BY a;
 a | count 
---+-------
 6 |   100
 7 |   100
 8 |   100
 9 |   100
(4 rows)

CREATE TABLE h(a INT);
INSERT INTO h SELECT g % 10 FROM generate_series(1, 1000) g;
-- insert into ducklake from an heap table
INSERT INTO t SELECT * FROM h;
SELECT COUNT(*) FROM t;
 count 
-------
  2000
(1 row)

DROP TABLE t;
DROP TABLE h;
-- empty table
CREATE TABLE empty(a INT) USING ducklake;
SELECT COUNT(*) FROM empty;
 count 
-------
     0
(1 row)

DROP TABLE empty;
